<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro" name="mobile_robot">
    <!-- COLOR -->
    <material name="white">
        <color rgba="1 1 1 1" />
    </material>

    <material name="black">
        <color rgba="0 0 0 1" />
    </material>

    <material name="blue">
        <color rgba="0 0 1 1"/>
    </material>

    <!-- BASE -->
    <link name="base_link">
    </link>

    <!-- BODY LINK -->
    <link name="body">
        <visual>
            <geometry>
                <box size="0.24 0.3 0.05"/>
            </geometry>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <material name="white"/>
        </visual>
        <collision>
            <geometry>
                <box size="0.24 0.3 0.05"/>
            </geometry>
        </collision>
        <inertia
            ixx="0.00025" ixy="0" ixz="0"
            iyy="0.00025" iyz="0"
            izz="0.00025"/>
    </link>

    <joint name="body_joint" type="fixed">
        <parent link="base_link"/>
        <child link="body"/>
        <origin xyz="0 0 0.105" rpy="0 0 0"/>
    </joint>

    <!-- LEFT WHEEL LINK -->
    <link name="left_wheel">
        <visual>
            <geometry>
                <cylinder length="0.04" radius="0.0435"/>
            </geometry>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/>
            <material name="black"/>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.04" radius="0.0435"/>
            </geometry>
        </collision>
        <inertia
            ixx="0.0001" ixy="0" ixz="0"
            iyy="0.0001" iyz="0"
            izz="0.0001"/>
    </link>

    <joint name="left_wheel_joint" type="continuous">
        <parent link="body"/>
        <child link="left_wheel"/>
        <axis xyz="0 1 0"/>
        <origin xyz="0 0.17 -0.0615"/>
    </joint>

    <!-- RIGHT WHEEL LINK -->
    <link name="right_wheel">
        <visual>
            <geometry>
                <cylinder length="0.04" radius="0.0435"/>
            </geometry>
            <origin xyz="0 0 0" rpy="1.5708 0 0"/>
            <material name="black"/>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.04" radius="0.0435"/>
            </geometry>
        </collision>
        <inertia
            ixx="0.0001" ixy="0" ixz="0"
            iyy="0.0001" iyz="0"
            izz="0.0001"/>
    </link>

    <joint name="right_wheel_joint" type="continuous">
        <parent link="body"/>
        <child link="right_wheel"/>
        <axis xyz="0 1 0"/>
        <origin xyz="0 -0.17 -0.0615"/>
    </joint>

    <!-- LIDAR LINK -->
    <link name="lidar">
        <visual>
            <geometry>
                <cylinder length="0.0737" radius="0.0425"/>
            </geometry>
            <origin xyz="0 0 0"/>
            <material name="blue"/>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.0737" radius="0.0425"/>
            </geometry>
        </collision>
        <inertia
            ixx="0.00005" ixy="0" ixz="0"
            iyy="0.00005" iyz="0"
            izz="0.00005"/>
    </link>

    <joint name="lidar_joint" type="fixed">
        <parent link="body"/>
        <child link="lidar"/>
        <origin xyz="0 0 0.03685"/>
    </joint>

    <!-- CAMERA LINK -->
    <link name="camera">
        <visual>
            <geometry>
                <box size="0.025 0.023 0.009"/>
            </geometry>
            <origin xyz="0 0 0" rpy="0 1.5708 0"/>
            <material name="blue"/>
        </visual>
        <collision>
            <geometry>
                <box size="0.025 0.023 0.009"/>
            </geometry>
        </collision>
        <inertia
            ixx="0.000001" ixy="0" ixz="0"
            iyy="0.000001" iyz="0"
            izz="0.000001"/>
    </link>

    <joint name="camera_joint" type="fixed">
        <parent link="body"/>
        <child link="camera"/>
        <origin xyz="0.12 0 0"/>
    </joint>
</robot>
